react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+
+class RegistrationForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: ''
+    };
+
+    this.handleUsernameChange = this.handleUsernameChange.bind(this);
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleUsernameChange(event) {
+    this.setState({ username: event.target.value });
+  }
+
+  handlePasswordChange(event) {
+    this.setState({ password: event.target.value });
+  }
+
+  handleSubmit(event) {
+    alert('Sign up successful');
+    console.log(this.state);
+    event.preventDefault();
+  }
+
+  render() {
+    return (
+      <div>
+        <form onSubmit={this.handleSubmit}>
+          <label>
+            Username
+            <input type="text" value={this.state.username} onChange={this.handleUsernameChange} />
+            </label>
+          <label>
+            Password
+            <input type="password" value={this.state.password} onChange={this.handlePasswordChange} />
+            </label>
+          <input type="submit" value='Sign Up' onChange={this.handleSubmit} />
```

Comment:
No on change handler should be used on the submit button input. The `<form>`s `onSubmit` handler will handle that.